### PR TITLE
fix(Dropdown): infinite render onChange, positioning in flex

### PR DESCRIPTION
**Pull request by saurabhdaware**
Fixes 2 bugs
- [Inifinite render](https://razorpay.slack.com/archives/CMQ3RBHEU/p1676970701130639?thread_ts=1676869625.828389&cid=CMQ3RBHEU)
- [positioning in display flex container](https://razorpay.slack.com/archives/CMQ3RBHEU/p1676871892270009?thread_ts=1676869625.828389&cid=CMQ3RBHEU)

### CODE REVIEW

File: packages/blade/src/components/Dropdown/Dropdown.stories.tsx [hunk 1]
```diff
@@ -510,13 +510,15 @@ export const WithValueDisplay = (args: AllDropdownProps): JSX.Element => {
...
     <BaseBox minHeight={300}>
-      <Text>Selected Values: {dropdownValues}</Text>
+      <Text>Selected Values: {JSON.stringify(dropdownValues)}</Text>
       <BaseBox marginTop="spacing.5" />
       <Dropdown selectionType={selectionType}>
         <SelectInput
           label="Select Action"
-          onChange={({ values }) => {
-            setDropdownValues(values.join(', '));
+          onChange={({ name, values }) => {
```
Comment (saurabhdaware) [inline packages/blade/src/components/Dropdown/Dropdown.stories.tsx:518]: it was working when state was string but it was breaking when state was object. Changed it in the story as well so we catch these issues.

File: packages/blade/src/components/Input/SelectInput/SelectInput.tsx [hunk 2]
```diff
@@ -66,7 +66,8 @@ const _SelectInput = (
 
   React.useEffect(() => {
     onChange?.({ name: props.name, values: value.split(', ') });
-  }, [value, onChange, props.name]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [value, props.name]);
```
Comment (saurabhdaware) [inline packages/blade/src/components/Input/SelectInput/SelectInput.tsx:70]: Having `onChange` in this array was making the function trigger on parent re-renders which was causing infinite re-renders.

Comment (sudhakar-selva) [reply packages/blade/src/components/Input/SelectInput/SelectInput.tsx:70]: [useEvent](https://github.com/reactjs/rfcs/blob/useevent/text/0000-useevent.md) hook might be useful here

Comment (saurabhdaware) [reply packages/blade/src/components/Input/SelectInput/SelectInput.tsx:70]: yup

File: packages/blade/src/components/Dropdown/Dropdown.stories.tsx [hunk 3]
```diff
@@ -720,4 +722,52 @@ WithRefUsage.args = {
...
+            <ActionList surfaceLevel={surfaceLevel}>
+              <ActionListItem title={title} value={value} />
+              <ActionListItem title="Primer" value="primer" />
+              <ActionListItem title="Geist" description="by Vercel" value="geist" />
+              <ActionListItem title="Airbnb Design" value="airbnb" />
+            </ActionList>
+          </DropdownOverlay>
+        </Dropdown>
+      </BaseBox>
+      <BaseBox flex={1}>
```
Comment (divyanshu013) [inline packages/blade/src/components/Dropdown/Dropdown.stories.tsx:751]: Minor: can perhaps add some spacing in between

**Review by divyanshu013 — APPROVED** [review]
Looks good to me.

File: packages/blade/src/components/Dropdown/Dropdown.stories.tsx (hunk 3 above)
Comment (saurabhdaware) [reply packages/blade/src/components/Dropdown/Dropdown.stories.tsx:751]: done

**Comment by saurabhdaware** [comment]
@kamleshchandnani can we merge this?

**Review by kamleshchandnani — COMMENTED** [review]
overall LGTM, one thing I would recommend is that
1. Can we explicitly add stories for controlled vs uncontrolled?
2. What are the remaining props of dropdown/select that we aren't part of the storybook we shall add all of them.

**Comment by saurabhdaware** [comment]
> Can we explicitly add stories for controlled vs uncontrolled?

We don't exactly have fully uncontrolled dropdown right now. For that we'll need `useDropdown` hook that can give functions like `selectOption` (this can be used for that keyboard bindings usecase we had discussed). 

We do have `isDefaultSelected` prop that sets default selection (I have added it in #1020. It was missing)

> What are the remaining props of dropdown/select that we aren't part of the storybook we shall add all of them.

Added it here #1020 (this is a bit of temporary solution though. not fully satisfied with docs right now. I'll fix it all when I am adding Button trigger and other things to docs)

**Pull request merged**